src/main/java/net/neoforged/neoforge/transfer/handlers/templates/contexts/CreativePlayerItemContext.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.handlers.templates.contexts;
+
+import net.minecraft.world.entity.player.Player;
+import net.neoforged.neoforge.transfer.ResourceHandlerUtil;
+import net.neoforged.neoforge.transfer.handlers.wrappers.items.PlayerInventoryWrapper;
+import net.neoforged.neoforge.transfer.resources.ItemResource;
+import net.neoforged.neoforge.transfer.transaction.TransactionContext;
+
+public class CreativePlayerItemContext extends StaticItemContext {
+    protected final PlayerInventoryWrapper handler;
+
+    public CreativePlayerItemContext(ItemResource resource, int amount, Player player) {
+        super(resource, amount);
+        this.handler = PlayerInventoryWrapper.of(player);
+    }
+
+    @Override
+    public int insert(ItemResource resource, int amount, TransactionContext transaction) {
+        if (ResourceHandlerUtil.isEmpty(resource, amount)) return 0;
+        boolean wasFound = ResourceHandlerUtil.contains(handler, resource);
+        return wasFound ? amount : handler.insert(resource, 1, transaction);
```

Comment:
This is not correct in case the resource was not found but more than 1 is being inserted. This sounds better:
```suggestion
        // Make sure that the player has at least one of the resource, but otherwise accept any insertion.
        if (!ResourceHandlerUtil.contains(handler, resource)) {
            handler.insert(resource, 1, transaction);
        }
        return amount;
```